### PR TITLE
CTCP-2783: Declaration Summary Page

### PR DESCRIPTION
**Pull request by Reece-Carruthers**
https://jira.tools.tax.service.gov.uk/browse/CTCP-2783

### CODE REVIEW

**Comment by Reece-Carruthers** [comment]
rebuild

File: app/controllers/VehicleRegistrationCountryController.scala [hunk 1]
```diff
@@ -45,32 +45,33 @@ class VehicleRegistrationCountryController @Inject() (
...
-  def onPageLoad(arrivalId: ArrivalId, mode: Mode): Action[AnyContent] = actions.requireData(arrivalId).async {
+  def onPageLoad(arrivalId: ArrivalId, transportMeansIndex: Index, mode: Mode): Action[AnyContent] = actions.requireData(arrivalId).async {
     implicit request =>
       referenceDataService.getCountries() map {
         countries =>
           val form = formProvider(countries)
-          val preparedForm = request.userAnswers.get(VehicleRegistrationCountryPage) match {
+          val preparedForm = request.userAnswers.get(VehicleRegistrationCountryPage(transportMeansIndex)) match {
             case None        => form
             case Some(value) => form.fill(Country(value, "test")) // TODO: Fix this change back to country
```
Comment (MatthewAlexanderBeck) [inline app/controllers/VehicleRegistrationCountryController.scala:55]: Can this todo be done now?

File: app/controllers/VehicleRegistrationCountryController.scala [hunk 2]
```diff
@@ -45,32 +45,33 @@ class VehicleRegistrationCountryController @Inject() (
...
             .bindFromRequest()
             .fold(
-              formWithErrors => Future.successful(BadRequest(view(formWithErrors, countries, request.userAnswers.mrn, arrivalId, mode))),
+              formWithErrors => Future.successful(BadRequest(view(formWithErrors, countries, request.userAnswers.mrn, arrivalId, transportMeansIndex, mode))),
               value =>
                 for {
-                  updatedAnswers <- Future.fromTry(request.userAnswers.set(VehicleRegistrationCountryPage, value.code)) // TODO: Fix this change back to country
-                  _              <- sessionRepository.set(updatedAnswers)
+                  updatedAnswers <- Future
+                    .fromTry(request.userAnswers.set(VehicleRegistrationCountryPage(transportMeansIndex), value.code)) // TODO: Fix this change back to country
```
Comment (MatthewAlexanderBeck) [inline app/controllers/VehicleRegistrationCountryController.scala:73]: same here maybe?

File: test/viewModels/UnloadingFindingsViewModelSpec.scala [hunk 3]
```diff
@@ -122,123 +148,99 @@ class UnloadingFindingsViewModelSpec extends SpecBase with AppWithDefaultMockFix
...
-        val viewModelProvider = new UnloadingFindingsViewModelProvider()
-        val result            = viewModelProvider.apply(userAnswers)
-        val section           = result.additionalComments
-
-        section.sectionTitle.value mustBe "Additional comments"
-        section.rows.size mustBe 1
-        section.addAnotherLink must not be defined
-      }
-    }
+//    "must render item section" - {
```
[the remark this answers is not in the document]
Comment (Reece-Carruthers) [reply test/viewModels/UnloadingFindingsViewModelSpec.scala:207]: Commented out as we don't need that section for this ticket any more. But will be used for CTCP-2784 so just left in in commented so we can resuse later

[the remark this answers is not in the document]
Comment (MatthewAlexanderBeck) [reply test/viewModels/UnloadingFindingsViewModelSpec.scala:207]: Ah cool, fine with that for reuse

**Review by MatthewAlexanderBeck — DISMISSED** [review]

**Review by kenneth-rayner — DISMISSED** [review]

**Review by kenneth-rayner — DISMISSED** [review]

**Review by kenneth-rayner — APPROVED** [review]

**Pull request merged**